transaction/std/src/memo_builder/mod.rs

```diff
@@ -34,15 +38,15 @@ pub trait MemoBuilder: Debug {
         value: u64,
         recipient: &PublicAddress,
         memo_context: MemoContext,
-    ) -> Result<Option<MemoPayload>, NewMemoError>;
+    ) -> Result<MemoPayload, NewMemoError>;
```

Comment:
I changed this because in the new world it doesn't make sense for a memo builder to sometimes omit the memo and sometimes not -- the block version controls if the memo is present or not, and the memo builder can just write `MemoPayload::default` if they don't want to write any content. This also means `NoMemoBuilder`goes away and is replaced by `EmptyMemoBuilder` everywhere it was used, since it will do the right thing based on Block version.